network-mux/src/Network/Mux.hs

```diff
@@ -126,34 +141,38 @@ muxStart peerid app bearer = do
       :: StrictTVar m Int
       -> PerMuxSharedState ptcl m
       -> ptcl
-      -> m [m ()]
+      -> m [(m (), String)]
     mpsJob cnt pmss mpdId = do
 
-        initiatorChannel <- muxChannel pmss
+        initiatorChannel <- muxChannel tracer
+                              pmss
                               (AppProtocolId mpdId)
                               ModeInitiator
                               cnt
 
-        responderChannel <- muxChannel pmss
+        responderChannel <- muxChannel tracer
+                              pmss
                               (AppProtocolId mpdId)
                               ModeResponder
                               cnt
 
         return $ case app of
-          MuxInitiatorApplication initiator -> [ initiator peerid mpdId initiatorChannel >> mpsJobExit cnt ]
-          MuxResponderApplication responder -> [ responder peerid mpdId responderChannel >> mpsJobExit cnt ]
-          MuxInitiatorAndResponderApplication initiator responder
-                                            -> [ initiator peerid mpdId initiatorChannel >> mpsJobExit cnt
-                                               , responder peerid mpdId responderChannel >> mpsJobExit cnt
-                                               ]
+          MuxInitiatorApplication initiator ->
+            [ (initiator peerid mpdId initiatorChannel >> mpsJobExit cnt , show mpdId ++ " Initiator")]
```

Comment:
these lines might exceed now 80 chars

Reply:
The line limit is 100 chars, but yes exceeds it. Will change.